mqtt/mqtt-broker/src/persist.rs

```diff
@@ -121,6 +124,53 @@ impl FileFormat for BincodeFormat {
         });
         let state = bincode::deserialize_from(decoder)
             .context(ErrorKind::Persist(ErrorReason::Deserialize))?;
+
+        Ok(state)
+    }
+
+    fn store<W: Write>(&self, writer: W, state: BrokerState) -> Result<(), Self::Error> {
+        let encoder = GzEncoder::new(writer, Compression::default());
+        fail_point!("bincodeformat.store.serialize_into", |_| {
+            Err(Error::from(ErrorKind::Persist(ErrorReason::Deserialize)))
+        });
+        bincode::serialize_into(encoder, &state)
+            .context(ErrorKind::Persist(ErrorReason::Serialize))?;
+        Ok(())
+    }
+}
+
+use mqtt3::proto::*;
+
+/// A simple format based on Bincode and serde
+#[derive(Clone, Debug, Default)]
+pub struct DeDupeFormat;
```

Comment:
Well, I think it's going to be the only production implementation. So no need to mention DeDuper part